iml-agent/src/action_plugins/stratagem/server.rs

```diff
@@ -126,26 +144,56 @@ fn has_flists<T: HaveFlist>(xs: &[T]) -> bool {
     xs.iter().any(HaveFlist::have_flist)
 }
 
+type MailboxFiles = Vec<(PathBuf, String)>;
+
 /// Given a results.json
 /// Returns all the directories that contain fid files.
-pub fn get_fid_dirs(results: &StratagemResult) -> Vec<String> {
-    results
+pub fn get_mailbox_files(
+    base_dir: &str,
+    stratagem_data: &StratagemData,
+    stratagen_result: &StratagemResult,
+) -> MailboxFiles {
+    stratagen_result
         .group_counters
         .iter()
         .filter(|x| has_flists(&x.counters))
-        .map(|x| &x.counters)
+        .map(|group| {
+            group
+                .counters
+                .iter()
+                .enumerate()
+                .filter(|(_, x)| HaveFlist::have_flist(x))
+                .map(move |(idx, counter)| {
+                    let group = stratagem_data
+                        .get_group_by_name(&group.name)
+                        .expect("did not find group by name");
+
+                    eprintln!("idx {:?}", idx);
+                    eprintln!("group {:?}", group);
+
+                    let rule = group
+                        .get_rule_by_idx(idx - 1)
+                        .expect("did not find rule by idx");
```

Comment:
Would it be helpful to indicate what index was used here?

Reply:
Sure